### PR TITLE
Add two desktop test cases and some reference files for DataContractSerializer test

### PR DESCRIPTION
**Pull request by JimboWei**
@shmao @huanwu Add two desktop test cases and some reference files for DataContractSerializer test. All the reference files are placed under **DesktopTestData** folder and using namespace of "**DesktopTestData**".

### CODE REVIEW

**Comment by shmao** [comment]
cc: @zhenlan

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 1]
```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
...
+        var dataContractSerializerSettings = new DataContractSerializerSettings()
+        {
+            DataContractResolver = new DesktopTestData.PrimitiveTypeResolver()
+            , IgnoreExtensionDataObject = false
+            , KnownTypes = null
+            , MaxItemsInObjectGraph = int.MaxValue
+            , PreserveObjectReferences = false
+        };
+
+        string baseline = "";
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Can you please set `baseline` to the expected payload?

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 2]
```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
...
+            , IgnoreExtensionDataObject = false
+            , KnownTypes = null
+            , MaxItemsInObjectGraph = int.MaxValue
+            , PreserveObjectReferences = false
+        };
+
+        string baseline = ""; 
+        var value = new DesktopTestData.ObjectContainer(new DesktopTestData.PrimitiveContainer());
+
+        var actual = SerializeAndDeserialize(value, baseline, dataContractSerializerSettings, null, true);
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Please set the last parameter to `false` to enable payload comparison.

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 3]
```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
...
+        // Throw Exception when verification failed
+        DesktopTestData.ComparisonHelper.CompareRecursively(value, actual);
+
+        var instance = new DesktopTestData.ObjectContainer(new DesktopTestData.PrimitiveContainer());
+
+        object result = DesktopTestData.DCRUtils.SingleRoundTripTest(
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: What's the difference between `SingleRoundTripTest` and `SerializeAndDeserialize`? Are they verifying different scenarios?

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 4]
```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
...
+                typeof(DesktopTestData.ObjectContainer)
+                , DesktopTestData.SerializerEnum.DataContractSerializer
+                , new DesktopTestData.PrimitiveTypeResolver()
+                )
+            , instance
+            );
+
+        DesktopTestData.ComparisonHelper.CompareRecursively(instance, result);
+
+        result = DesktopTestData.DCRUtils.SingleRoundtripPerEpisode(
```
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: What's the difference between `SingleRoundtripPerEpisode` and `SerializeAndDeserialize`? Are they verifying different scenarios?

**Comment by shmao** [comment]
@JimboWei Thanks for the PR. I wondered if we actually need all these files for the test infrastructure. I tried simplifying the tests a little bit. Would you mind taking a look at https://github.com/shmao/corefx/commit/958b29e5d9030f1a8ed17145bf1b45474edc6054 and see if the simplified version would work? Would they work for other tests that depend on the infrastructure?

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs (hunk 1 above)
Comment (JimboWei) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: The object has some fields which are using **dynamic generated value**( such as DateTime.Now and Guid.NewGuid() ),I will try to use hard coded value instead and set expected value to `baseline`.

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs (hunk 3 above)
Comment (JimboWei) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: They are same I will keep `SerializeAndDeserialize`.

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs (hunk 4 above)
Comment (JimboWei) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: I think they are similar. The `SingleRoundtripPerEpisode` wrapped a `XmlDictionaryWriter `when serialize object. I will update code to keep `SerializeAndDeserialize`.

**Comment by JimboWei** [comment]
@shmao The simplified version [shmao@`958b29e`](https://github.com/shmao/corefx/commit/958b29e5d9030f1a8ed17145bf1b45474edc6054) works very well. The other files are used to some test cases which I haven't added in this project. I'm reviewing and debugging the code to verify these cases, so that they can be added in this project.
Thanks for your review , and I know this commit is a little large, maybe I should only add test cases with their necessary reference files next time.  If you have any suggestions, please let me know.

**Comment by shmao** [comment]
@JimboWei The PR LGTM. Can you please address the following issues?

1. Complete the Contribution License Agreement at https://cla2.dotnetfoundation.org.
2. Rename the namespace and folder from `DesktopTestData` to `SerializationTestTypes`.
3. Resolve the conflicts between your branch and the master branch so that we can merge the PR. Please see below for the commands you may need,

From your work branch,
```
git fetch --prune --all
git merge --ff-only upstream/master 
(Then resolve any conflicts. It would be easier to resolve the conflicts in Visual Studio.)
```

**Comment by JimboWei** [comment]
Thanks @shmao . Due to my Email problem, I just finished signing the CLA. 
What I have done is below:
1. Rename the namespace and folder.
2. Format the code.
3. Resolve the conflicts.

Could you please help me review this commit ?

**Comment by shmao** [comment]
@JimboWei The tests LGTM. But the PR got conflicts again (probably due to other changes getting merged). Can you please resolve the conflicts again? Thanks!

**Comment by JimboWei** [comment]
@shmao I have merge the code and resolve the conflicts with @v-yanbgu after some communication. Could you please help me review this commit ?

**Comment by shmao** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release x64 Build and Test

**Comment by shmao** [comment]
@JimboWei Thanks for adding the tests.

**Pull request merged**